File: modules/ia_pharmacy/ios/ia_pharmacy/Sources/ia_pharmacy/IaSdkFlutterPharmacy.swift
import Flutter
import UIKit
import IACore
import IAPharmacy

@MainActor
public class IaSdkFlutterPharmacy: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(
        name: "de.ihreapotheken/sdk/pharmacyDetails",
        binaryMessenger: registrar.messenger(),
    )
    let instance = IaSdkFlutterPharmacy()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "register":
        IASDK.register([IASDKModule.pharmacyDetails])
        result(nil)
    default:
        result(FlutterMethodNotImplemented)
    }
  }
}

File: modules/ia_pharmacy/android/src/main/kotlin/de/ihreapotheken/appsdk/flutter/pharmacy/IaSdkFlutterPharmacy.kt
package de.ihreapotheken.appsdk.flutter.pharmacy

import de.ihreapotheken.sdk.integrations.api.IaSdk
import de.ihreapotheken.sdk.pharmacy.PharmacyModule
import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** IaSdkFlutterPharmacy */
class IaSdkFlutterPharmacy :
    FlutterPlugin,
    MethodCallHandler {
    // The MethodChannel that will the communication between Flutter and native Android
    //
    // This local reference serves to register the plugin with the Flutter Engine and unregister it
    // when the Flutter Engine is detached from the Activity
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(
            flutterPluginBinding.binaryMessenger,
            "de.ihreapotheken/sdk/pharmacyDetails",
        )
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(
        call: MethodCall,
        result: Result
    ) {
        if (call.method == "register") {
            IaSdk.register(PharmacyModule)
            result.success(null)
        } else {
            result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}
